{"name": "Test job cities from the velpitar website against Peviitor API Response", "status": "failed", "statusDetails": {"message": "AssertionError: An unknown error occurred", "trace": "get_job_details = ((['Manager Vânzări (Distribuitori) - București', 'Manager Vanzari IKA / Key Account Manager - București', 'Analist fi...r', 'Velpitar', 'Velpitar', 'Velpitar', ...], ['on-site', 'on-site', 'on-site', 'on-site', 'on-site', 'on-site', ...]))\n\n    @pytest.mark.regression\n    @pytest.mark.API\n    def test_velpitar_city_api(get_job_details):\n        allure.dynamic.title(f\"Test job cities from the {company_name} website against Peviitor API Response\")\n    \n        scraped_jobs_data, peviitor_jobs_data = get_job_details\n        with allure.step(\"Step 1: Get job cities and titles from the scraper\"):\n            job_cities_scraper = scraped_jobs_data[1]\n            job_titles_scraper = scraped_jobs_data[0]\n    \n        with allure.step(\"Step 2: Get job cities and titles from the Peviitor API\"):\n            job_cities_peviitor = peviitor_jobs_data[1]\n            job_titles_peviitor = peviitor_jobs_data[0]\n    \n        with allure.step(\"Step 3: Compare job cities from scraper response against Peviitor API Response\"):\n            allure.attach(f\"Expected Results: {job_cities_scraper}\", name=\"Expected Results\")\n            allure.attach(f\"Actual Results: {job_cities_peviitor}\", name=\"Actual Results\")\n>           TestUtils().check_job_cities(job_cities_scraper, job_cities_peviitor, job_titles_scraper, job_titles_peviitor)\n\ntests/Test_API_Suite/test_velpitar.py:56: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <tests.utils.TestUtils object at 0x7f38c9b59030>\nexpected_cities = [['Bucuresti'], ['Bucuresti'], ['Bucuresti'], ['Bucuresti'], ['Bucuresti'], ['Bucuresti'], ...]\nactual_cities = [['Brasov'], ['Chitila'], ['Giurgiu'], ['Giurgiu'], ['Iasi'], ['Iasi'], ...]\njob_titles_scraper = ['Manager Vânzări (Distribuitori) - București', 'Manager Vanzari IKA / Key Account Manager - București', 'Analist fina... - București', 'Analist financiar senior - București', 'Brand Manager - Bucuresti', 'Analist vanzari - Bucuresti', ...]\napi_job_titles = ['Gestionar depozit - Brasov', 'Electrician întreţinere şi reparaţii – Chitila', 'Mecanic/Lacătuș mecanic - Giurgiu', ...parații - Giurgiu', 'Gestionar depozit - Iasi', 'Absolvent Inginerie electrică, Electronică, Automatizări - Iași', ...]\n\n    def check_job_cities(self, expected_cities, actual_cities, job_titles_scraper, api_job_titles):\n        if not expected_cities:\n            msg = f\"Scraper is not grabbing any job cities\"\n            allure.step(msg)\n            raise AssertionError(msg)\n    \n        msg = \"An unknown error occurred\"\n    \n        # Check jobs from scraper against the peviitor api\n        scraper_actual_cities, scraper_job_titles = self.get_different_items(expected_cities, actual_cities, job_titles_scraper)\n    \n        if scraper_actual_cities:\n            msg = f\"Peviitor is missing job cities for the following job titles: {scraper_job_titles}\"\n            allure.step(msg)\n            raise AssertionError(msg)\n        else:\n            # Check jobs from peviitor against the scraper response\n            peviitor_actual_cities, peviitor_job_titles = self.get_different_items(actual_cities, expected_cities, api_job_titles)\n            if peviitor_actual_cities:\n                msg = f\"Peviitor is having extra job cities for the following job titles: {peviitor_job_titles}\"\n                allure.step(msg)\n                raise AssertionError(msg)\n    \n        allure.step(msg)\n>       assert expected_cities == actual_cities, msg\nE       AssertionError: An unknown error occurred\n\ntests/utils.py:151: AssertionError"}, "steps": [{"name": "Step 1: Get job cities and titles from the scraper", "status": "passed", "start": 1719119572313, "stop": 1719119572313}, {"name": "Step 2: Get job cities and titles from the Peviitor API", "status": "passed", "start": 1719119572313, "stop": 1719119572313}, {"name": "Step 3: Compare job cities from scraper response against Peviitor API Response", "status": "failed", "statusDetails": {"message": "AssertionError: An unknown error occurred\n", "trace": "  File \"/home/runner/work/RaresTestHub/RaresTestHub/tests/Test_API_Suite/test_velpitar.py\", line 56, in test_velpitar_city_api\n    TestUtils().check_job_cities(job_cities_scraper, job_cities_peviitor, job_titles_scraper, job_titles_peviitor)\n  File \"/home/runner/work/RaresTestHub/RaresTestHub/tests/utils.py\", line 151, in check_job_cities\n    assert expected_cities == actual_cities, msg\n"}, "attachments": [{"name": "Expected Results", "source": "9ff74801-395d-4304-8de3-840108946977-attachment.attach"}, {"name": "Actual Results", "source": "92929e27-2397-4b48-9c74-0f17ce6db3da-attachment.attach"}], "start": 1719119572313, "stop": 1719119572314}], "start": 1719119572313, "stop": 1719119572314, "uuid": "098d9af4-8c1b-4361-8ecc-1ce0779fd72c", "historyId": "8685014eae2f3b187ae484ad46e1a6b7", "testCaseId": "8685014eae2f3b187ae484ad46e1a6b7", "fullName": "Test_API_Suite.test_velpitar#test_velpitar_city_api", "labels": [{"name": "tag", "value": "API"}, {"name": "tag", "value": "regression"}, {"name": "parentSuite", "value": "Test_API_Suite"}, {"name": "suite", "value": "test_velpitar"}, {"name": "host", "value": "fv-az1486-994"}, {"name": "thread", "value": "2220-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "Test_API_Suite.test_velpitar"}]}